src/core/meshcop/border_agent.cpp

```diff
@@ -840,6 +842,24 @@ void BorderAgent::HandleDtlsTransportClosed(void)
     RestartAfterRemovingEphemeralKey();
 }
 
+void BorderAgent::SetEphemeralKeyFeatureEnabled(bool aEnabled)
+{
+    VerifyOrExit(mIsEphemeralKeyFeatureEnabled != aEnabled);
+    mIsEphemeralKeyFeatureEnabled = aEnabled;
+
+    if (!mIsEphemeralKeyFeatureEnabled)
+    {
+        // If the ePSKc feature is enabled, we call the clear function which
+        // will wait for the session to close if it is in active use before
+        // removing ephemeral key and unpublishing the service.
+        ClearEphemeralKey();
```

Comment:
the ephemeral key will not be cleared if there is an active commissioner connection, shall we force terminate the connection and clear ephemeral key?

```
Disconnect();
ClearEphemeralKey();
```

Reply:
As I see in the comment in `ClearEphemeralKey`:
```
        // If a commissioner connection is currently active, we'll
        // wait for it to disconnect or for the ephemeral key timeout
        // or `kKeepAliveTimeout` to expire before removing the key
        // and restarting the agent.
```
So I think the design is to wait for it to disconnect instead of disconnecting it proactively.

Reply:
I slightly prefer to disconnect and clear so that when the epskc feature is disabled, there is no use of the epskc anymore.

On the other hand, I'm also fine with waiting for disconnection since the connection was established when the epskc was still valid. In this case maybe update the doc for `otBorderAgentSetEphemeralKeyFeatureEnabled` to say that the in-use ephemeral key will be removed after there is no active connection?

Reply:
I agree with @sunytt; it may be better to disconnect. When we are disabling the ephemeral key feature fully, it makes sense to stop all its functions, so it may be better to disconnect any ongoing session using the ephemeral key.

Reply:
Thanks! I added the disconnect here with a check to ensure the session is connected with a ephemeral key: If there is an active normal session that doesn't connect with ephemeral key, and we disable the ephemeral key feature, then I think we shouldn't disconnect the session.